src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -181,6 +181,7 @@ public override Binder VisitMethodDeclaration(MethodDeclarationSyntax methodDecl
                     if (usage == NodeUsage.MethodBody)
                     {
                         method = method ?? GetMethodSymbol(methodDecl, resultBinder);
+                        Debug.Assert(method is not null);
```

Comment:
>Debug.Assert(method is not null); [](http://example.com/codeflow?start=24&length=33)

It looks like the constructor already asserts that. So, I am not sure what extra value we are going to get from asserting at the call sites.